homeassistant/components/esphome/entity.py

```diff
@@ -200,25 +201,6 @@ def __init__(
         self._attr_device_info = DeviceInfo(
             connections={(dr.CONNECTION_NETWORK_MAC, device_info.mac_address)}
         )
-        #
-        # If `friendly_name` is set, we use the Friendly naming rules, if
-        # `friendly_name` is not set we make an exception to the naming rules for
-        # backwards compatibility and use the Legacy naming rules.
-        #
-        # Friendly naming
-        # - Friendly name is prepended to entity names
-        # - Device Name is prepended to entity ids
-        # - Entity id is constructed from device name and object id
-        #
-        # Legacy naming
-        # - Device name is not prepended to entity names
-        # - Device name is not prepended to entity ids
-        # - Entity id is constructed from entity name
-        #
-        if not device_info.friendly_name:
-            return
-        self._attr_has_entity_name = True
-        self.entity_id = f"{domain}.{device_info.name}_{entity_info.object_id}"
```

Comment:
This shouldn't have been removed in the case of ESPHome.

As these IDs are already defined upstream. This was specifically added for ESPHome (as a matter of fact, the API libraries have been adjusted to enable this even).

This needs to be restored.